pydantic/main.py

```diff
@@ -369,7 +369,12 @@ def __setattr__(self, name: str, value: Any) -> None:
             _object_setattr(self, name, value)
             return
         elif self.model_config.get('frozen', None):
-            raise TypeError(f'"{self.__class__.__name__}" is frozen and does not support item assignment')
+            error: pydantic_core.InitErrorDetails = {
+                'type': 'frozen_instance',
+                'loc': ('__root__',),
```

Comment:
Is this really the loc we want? I almost think an empty tuple would be better, was hoping to get rid of the awkward references to `__root__` if possible. @samuelcolvin do you have thoughts?

Reply:
Also, given this happened in a call to `__setattr__` and you have the name, I think
```suggestion
                'loc': (name,),
```
might be better?

Reply:
I think `name` is better. Changed based on your suggestion.